test/framework/toolchain.py

```diff
@@ -709,7 +709,7 @@ def test_optarch_generic(self):
             tcs = {
                 'gompi': ('2018a', "-march=x86-64 -mtune=generic"),
                 'iccifort': ('2018.1.163', "-xSSE2 -ftz -fp-speculation=safe -fp-model source"),
-                'intel-compilers': ('2021.4.0', "-xSSE2 -fp-speculation=safe -fp-model precise"),
+                'intel-compilers': ('2021.4.0', "-march=x86-64 -mtune=generic -fp-speculation=safe -fp-model precise"),
```

Comment:
This is a bit too lax. It should test intel-compilers with and without oneapi enabled.
So perhaps skip the enforcing of oneapi=true below and have two intel-compilers, one pre-2022.2.0 (like the 2021.4.0 above) with -xSSE2 and one intel-compilers using version > 2022.2.0 to get oneapi auto enabled.
Not sure how to do that easy since you can't have two intel-compilers in this dict...

perhaps an array of single compiler dicts?

Reply:
Very good point for raising this: I've extended this test extensively in 848293d, and doing so actually uncovered a bug, which was fixed in 63099bf